spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/ObjectFactory.java

```diff
@@ -0,0 +1,20 @@
+package org.eclipse.dataspaceconnector.spi;
+
+import org.eclipse.dataspaceconnector.spi.system.Feature;
+
+/**
+ * Factory object that is used to generate instances of objects based on their class.
+ * One way to implement this is to generate a new instance on demand using reflection and dependency injection.
+ */
+@FunctionalInterface
+@Feature("edc:core:object-factory")
+public interface ObjectFactory {
+    /**
+     * Gets (and potentially creates) a new instance of a commandHandler
+     *
+     * @param clazz The object's class
+     * @return An instance of the object, potentially newly created, or null if none could be found.
```

Comment:
Would there ever be a case where null is returned? Shouldn't a runtime exception be thrown instead?